src/Arcus.Security.AzureFunctions/Extensions/IFunctionHostBuilderExtensions.cs

```diff
@@ -27,5 +28,23 @@ public static IFunctionsHostBuilder ConfigureSecretStore(this IFunctionsHostBuil
             functionsHostBuilder.Services.AddSecretStore(configureSecretStores);
             return functionsHostBuilder;
         }
+
+        /// <summary>
+        /// Configure an <see cref="ISecretProvider"/> in the application with a given set of stores configured in the given <paramref name="configureSecretStores"/>.
+        /// </summary>
+        /// <param name="functionsHostBuilder">The builder to append the secret store configuration to.</param>
+        /// <param name="configureSecretStores">The customization of the different target secret store sources to include in the final <see cref="ISecretProvider"/>.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="functionsHostBuilder"/> or <paramref name="configureSecretStores"/> is <c>null</c>.</exception>
+        public static IFunctionsHostBuilder ConfigureSecretStore(
+            this IFunctionsHostBuilder functionsHostBuilder,
+            Action<FunctionsHostBuilderContext, IConfiguration, SecretStoreBuilder> configureSecretStores)
+        {
+            Guard.NotNull(functionsHostBuilder, nameof(functionsHostBuilder), "Requires a functions host builder to add the secret store");
+            Guard.NotNull(configureSecretStores, nameof(configureSecretStores), "Requires a function to configure the secret store with potential secret providers");
+
+            FunctionsHostBuilderContext context = functionsHostBuilder.GetContext();
```

Comment:
Not sure if this should be called here or inside the `AddSecretStore`. The call would be delayed, and outside of the context. I didn't see much context-specifics when inspecting `.GetContext` but not sure if we should take the risk of moving it inside the lambda.

Reply:
I would keep it as is.